B 22742 int

CoryKleinjanCACI opened the pull request:
## [B-22742](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1068589)

## Summary

This task adds the ICRT and IUCRT service items to the update_estimated_pricing stored procedure

More specifically this addresses an issue where the estimated pricing was not matching the actual pricing.

NOTE: There is an outstanding issue where using decimal inches calculates incorrectly. This will need a seperate issue created.

## Verification Steps for Reviewers


### How to test

1. Access the milmove app
2. Login as a customer
3. Create an international move with an hhg
4. Login as SC and submit move details for the move you created
5. Login as TOO and approve the move you created
6. Login to Prime Sim and add both an ICRT and IUCRT service item to the hhg shipment
7. Fill in all details for the ICRT and IUCRT
8. Update the shipments pickup dates, delivery dates and estimated weight.
9. Validate the estimated price is being stored in the mto_service_items table under 'pricing_estimate' 
10. Navigate to the Prime Simulator.
11. Create a payment request that includes the ICRT and IUCRT service items.
12. Login as a TIO
13. Navigate to the move and validate the payment request you created. 
14. Validate the the actual pricing matches the estimated pricing.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).


### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

MInthavongsay commented on migrations/app/ddl_migrations/ddl_functions/20250221213413_fn_update_service_item_pricing.up.sql:
```diff
@@ -108,15 +126,30 @@ BEGIN
 
                 IF length IS NOT NULL AND height IS NOT NULL AND width IS NOT NULL THEN
-                    estimated_price := ROUND((escalated_price * (((length/1000) * (width/1000) * (height/1000)) / 100)), 2) * 100;
+                    cubic_feet := ((length/1000) * (width/1000) * (height/1000)) / 1728;
```
Numbers like 1728, 4.0, 100000.00 something we can insert into the `application_parameters` table? Then use function `get_application_parameter_value` to retrieve it.

CoryKleinjanCACI replied:
Updated to use application_parameters for standalone_crate_cap and external_crate_minimum. I left 1728 as that is just part of the mathematical formula to get cubic feet from inches, so it will never change.

ajlusk commented on migrations/app/ddl_migrations/ddl_functions/20250221213413_fn_update_service_item_pricing.up.sql:
```diff
@@ -34,6 +34,11 @@ DECLARE
+    external BOOLEAN;
+    cubic_feet NUMERIC;
+    standalone_crate_cap;
+    external_crate_minimum;
```
My local doesn't seem to like the lack of type declarations when restarting the server. I'm getting `pq: missing data type declaration at or near ";"` which goes away when I give these a type.

It didn't happen when I first checked out the branch or the first startup after deleting the DB container but does happen on any other startup of the server.

Probably good practice to include a type declaration anyway.

CoryKleinjanCACI replied:
whoops. those should definitely have types.

MInthavongsay replied:
Getting an error while creating the SIT. 
![Screenshot 2025-05-09 at 11 28 34 AM](https://github.com/user-attachments/assets/f3646899-a307-45c9-9cd4-d50ab3ddf1e6)

MInthavongsay commented on migrations/app/ddl_migrations/ddl_functions/20250221213413_fn_update_service_item_pricing.up.sql:
```diff
@@ -49,6 +54,16 @@ BEGIN
         RETURN;
     END IF;
 
+    SELECT parameter_value
```
This is good for now querying application_parameters table directly. If the opportunity arises -- using another BL -- may want to make a mental note to come back and use `get_application_parameter_value` when the function is finally merged into MAIN.

MInthavongsay reviewed (dismissed):
Looks good.

ajlusk reviewed (approved):
~Looks good after changes. I remember thinking the pricing seemed awfully high when I reviewed the first PR.~

Agree with that ideally it should use the pre-existing application parameter function but not a blocker.

ajlusk reviewed (dismissed):
Sorry for the double review but I missed this last time since it wasn't in the diffs.

antgmann reviewed (commented):
Not seeing any client test failures on my local, but I am seeing the server test failure gitlab is reporting.
<img width="722" alt="image" src="https://github.com/user-attachments/assets/c18960f9-bbaf-4fb7-a16e-1b921bd6e8f9" />

CoryKleinjanCACI commented:
> Not seeing any client test failures on my local, but I am seeing the server test failure gitlab is reporting. <img alt="image" width="722" src="https://private-user-images.githubusercontent.com/148906585/442839531-c18960f9-bbaf-4fb7-a16e-1b921bd6e8f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNjg4MjAsIm5iZiI6MTc0NzA2ODUyMCwicGF0aCI6Ii8xNDg5MDY1ODUvNDQyODM5NTMxLWMxODk2MGY5LWJiYWYtNGZiNy1hMTZlLTFiOTIxYmQ2ZThmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxMlQxNjQ4NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03N2JmNDY3YTJlZDU1NTQxMzkxY2I0YjE1YWE3NDA1Y2JkZTkyZWUzMjZjYjM5ZTY1OWE3MmRmZmY3NDRjYmY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xmNhMfAYKRnFoqs_UcM3-Xrj-Es22RjJ0qhQ760-oTo">

Looks like this was caused by the same issue as some of the others. Calls to calculate_escalated_price werent updated when the number of parameters were changed, so it was calling an outdated function. I removed the outdated functions and updated this call now. Probably not explicitly in scope for this task, but it's not possible to get the tests passing without this fix so I went ahead and fixed it.

MInthavongsay reviewed (approved):
Looks good. Server tests are now passing.

antgmann reviewed (approved):
Reapproving
